botnim/vector_store/vector_store_es.py

```diff
@@ -296,4 +287,46 @@ def verify_document_metadata(self, index_name: str, document_id: str) -> Dict:
             return result['_source'].get('metadata', {})
         except Exception as e:
             logger.error(f"Failed to verify metadata for document {document_id}: {str(e)}")
-            return {}
\ No newline at end of file
+            return {}
+
+    def create_search_tool(self, bot_name: str, context_name: str, environment: str) -> Dict:
+        """Creates an Elasticsearch vector search tool configuration for a specific context"""
+        
+        # Load the bot's config to get context details
+        config_path = Path(self.config_dir) / 'config.yaml'
+        with open(config_path) as f:
+            config = yaml.safe_load(f)
+        
+        # Find the specific context configuration to get its slug
+        context_config = next(
+            (ctx for ctx in config.get('context', []) if ctx['name'] == context_name),
+            {'slug': context_name.lower().replace(' ', '_')}  # fallback to sanitized name
+        )
+        
+        # Use slugs for the tool name
+        tool_name = f"ElasticVectorSearch_{bot_name}_{context_config['slug']}"
```

Comment:
This is ignoring the environment - creating the same tool's name for production and staging.

Reply:
good catch. fixed.